container/mesos/client.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesos
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"sync"
+)
+
+const (
+	slaveURL               = "http://localhost:5051"
+	containersHTTPEndpoint = "/containers"
+	slaveStateHTTPEndpoint = "/state"
+	timeout                = 10 * time.Second
+)
+
+var (
+	mesosClientOnce sync.Once
+	mesosClient     *client
+)
+
+type client struct {
+	hc  *http.Client
+	url string
+}
+
+type mesosSlaveClient interface {
+	ContainerInfo(id string) (*containerInfo, error)
+}
+
+type containerInfo struct {
+	cntr   *mesosContainer
+	labels map[string]string
+}
+
+// Client is an interface to query mesos slave http endpoints
+func Client() (mesosSlaveClient, error) {
+	mesosClientOnce.Do(func() {
```

Comment:
why not just a proper constructor (e.g. something like `func New`)? this doesn't seem like a pattern people commonly use for Go...

Reply:
I saw this method in all the containers already present in Cadvisor, so followed their style here.